src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheClient.java

```diff
@@ -60,6 +61,7 @@ public boolean containsActionResult(ActionKey actionKey) {
 
   public void captureFile(Path src, Digest digest, boolean isActionCache) throws IOException {
     Path target = toPath(digest.getHash(), isActionCache);
+    ensureParentExists(target);
```

Comment:
Path has a method called createDirectoryAndParents. Use that instead?

Reply:
Somehow had missed it. Changed to using that.